calyx/src/passes/well_formed.rs

```diff
@@ -120,68 +130,15 @@ where
     Ok(())
 }
 
-fn same_binding(
-    name_out: &ir::Id,
-    binding_out: &ir::Binding,
-    binding_in: &ir::Binding,
-) -> CalyxResult<()> {
-    if binding_out.len() != binding_in.len() {
-        return Err(Error::malformed_control(format!(
-            "unmatching binding sizes, expected {}, provided {}",
-            binding_out.len(),
-            binding_in.len()
-        )));
-    }
-
-    binding_out.iter().zip(binding_in.iter()).try_for_each(
-        |((id_out, value_out), (id_in, value_in))| {
-            if id_out == id_in && value_out == value_in {
-                Ok(())
-            } else {
-                Err(Error::malformed_control(
-                    format!("unmatching binding values for {name_out}, expected {id_out} to be {value_out}, instead got {value_in}"),
-                ))
-            }
-        },
-    )
-}
-
 fn same_type(proto_out: &CellType, proto_in: &CellType) -> CalyxResult<()> {
-    match (proto_out, proto_in) {
```

Comment:
@paili0628 `CellType` implements `Eq` and therefore does not require manually equality comparison. Does this make sense?